src/list.cpp

```diff
@@ -117,126 +144,246 @@ void List::verify_in_transaction() const
 size_t List::size() const
 {
     verify_attached();
-    return m_link_view->size();
+    return m_link_view ? m_link_view->size() : m_table->size();
+}
+
+size_t List::to_table_ndx(size_t row) const noexcept
+{
+    return m_link_view ? m_link_view->get(row).get_index() : row;
 }
 
-RowExpr List::get(size_t row_ndx) const
+PropertyType List::get_type() const
 {
     verify_attached();
+    return m_link_view ? PropertyType::Object
+                       : ObjectSchema::from_core_type(*m_table->get_descriptor(), 0);
+}
+
+namespace {
+template<typename T>
+auto get(Table& table, size_t row)
+{
+    return table.get<T>(0, row);
+}
+
+template<>
+auto get<RowExpr>(Table& table, size_t row)
+{
+    return table.get(row);
+}
+}
+
+template<typename T>
+T List::get(size_t row_ndx) const
+{
     verify_valid_row(row_ndx);
-    return m_link_view->get(row_ndx);
+    return ::get<T>(*m_table, to_table_ndx(row_ndx));
 }
 
-size_t List::get_unchecked(size_t row_ndx) const noexcept
+template RowExpr List::get(size_t) const;
+
+template<typename T>
+size_t List::find(T const& value) const
 {
-    return m_link_view->get(row_ndx).get_index();
+    verify_attached();
+    return m_table->find_first(0, value);
 }
 
-size_t List::find(ConstRow const& row) const
+template<>
+size_t List::find(RowExpr const& row) const
```

Comment:
[GCC 5.4 appears to take issue with this](https://ci.realm.io/blue/organizations/jenkins/realm%2Frealm-object-store/detail/PR-512/4/pipeline/35):
```
/src/list.cpp:193:39: error: specialization of 'template<class T> size_t realm::List::find(const T&) const' in different namespace [-fpermissive]
 size_t List::find(RowExpr const& row) const
                                       ^
In file included from src/list.cpp:19:0:
src/list.hpp:85:12: error:   from definition of 'template<class T> size_t realm::List::find(const T&) const' [-fpermissive]
     size_t find(T const& value) const;
```

I guess this is because the file has `using namespace realm;` rather than its contents being enclosed in `namespace realm { … }`?